### PR TITLE
Fixes to run under Python 3 (and thus under ESX 6.5 GA) 

### LINKED ISSUE #666: vSphere 6.5 readiness
[vSphere 6.5](https://blogs.vmware.com/vsphere/2016/10/introducing-vsphere-6-5.html) is on it's way. This issue is to track and verify vSphere docker volume service works out of box on ESX 6.5

- ESX 6.5 ships with Python 3 only. Make sure ESX components (vmdkops and admin CLI) are Python 3 compatible
- Install VIB and run CI tests with ESX 6.5
- Add ESX 6.5 to CI test matrix.

CC/ @msterin @shuklanirdesh82

### PR DESCRIPTION
**Pull request by msterin**
This fixes remaining python3 related issues  in esx_service. 

Fixes #666 

Testing:
1. local test:  `make test-esx` on my ESX 6.5 testbed. 
2. CI for python2 (in flight)
3. Manual create/delete/mount of volumes on VSAN 6.5

We will have Python3/ESX 6.5 in the  CI/CD testbed  per @kerneltime , early next week.

Test report from my local test is as follows: 
```shell
msterin-m01:docker-volume-vsphere msterin$ ESX=$ESX_65 VM=$VM_65 make deploy-esx test-esx
/Applications/Xcode.app/Contents/Developer/usr/bin/make --directory=vmdk_plugin deploy-esx
../misc/scripts/deploy-tools.sh cleanesx "10.192.119.197" esx-vmdkops-service

=> Cleaning up on ESX 10.192.119.197
=> Fri Nov 18 19:21:12 PST 2016

../misc/scripts/deploy-tools.sh deployesx "10.192.119.197" "../build/vmware-esx-vmdkops-0.8.1.c0b337b.vib"

=> Deploying to ESX root@10.192.119.197
=> Fri Nov 18 19:21:15 PST 2016

Installation Result:
   Message: WARNING: Only live system was updated, the change is not persistent.
   Reboot Required: false
   VIBs Installed: VMWare_bootbank_esx-vmdkops-service_0.8.1.c0b337b-0.0.1
   VIBs Removed: 
   VIBs Skipped: 
vmdkops-opsd is running pid=275182
/Applications/Xcode.app/Contents/Developer/usr/bin/make --directory=vmdk_plugin test-esx

=> Running target test-esx
=> Fri Nov 18 19:21:22 PST 2016

/Applications/Xcode.app/Contents/Developer/usr/bin/make  --directory=../esx_service test-esx
Copying files to 10.192.119.197:/tmp/vmdk_ops_unittest18380 and running .py unittests ...
ssh  -kTax -o StrictHostKeyChecking=no root@10.192.119.197 'mkdir -p /tmp/vmdk_ops_unittest18380'
scp  -r -q -o StrictHostKeyChecking=no ./cli/vmdkops_admin_sanity_test.py ./cli/vmdkops_admin_test.py ./utils/auth_data_test.py ./utils/cli_table_test.py ./utils/vsan_info_test.py ./vmdk_ops_test.py ./vsan_policy_test.py __init__.py vmdk_ops.py volume_kv.py vsan_policy.py cli/vmdkops_admin.py utils/__init__.py utils/auth.py utils/auth_data.py utils/auth_data_const.py utils/cli_table.py utils/convert.py utils/kvESX.py utils/log_config.py utils/threadutils.py utils/vmdk_utils.py utils/vsan_info.py root@10.192.119.197:/tmp/vmdk_ops_unittest18380
ssh  -kTax -o StrictHostKeyChecking=no root@10.192.119.197 \
		'for i in /tmp/vmdk_ops_unittest18380/'*'_test.py ; \
				do echo Running unit tests in $i... ; python $i ; \
				status=$((status + $?)) ;\
		done; \
		exit $status'
Running unit tests in /tmp/vmdk_ops_unittest18380/auth_data_test.py...
.........
----------------------------------------------------------------------
Ran 9 tests in 0.022s

OK
Running unit tests in /tmp/vmdk_ops_unittest18380/cli_table_test.py...
......
----------------------------------------------------------------------
Ran 6 tests in 0.001s

OK
Running unit tests in /tmp/vmdk_ops_unittest18380/vmdk_ops_test.py...
ss........ss.
----------------------------------------------------------------------
Ran 13 tests in 9.320s

OK (skipped=4)
Running unit tests in /tmp/vmdk_ops_unittest18380/vmdkops_admin_sanity_test.py...
...
----------------------------------------------------------------------
Ran 3 tests in 2.384s

OK
Running unit tests in /tmp/vmdk_ops_unittest18380/vmdkops_admin_test.py...
...........................................
----------------------------------------------------------------------
Ran 43 tests in 9.709s

OK
Version: 0.8.1.c0b337b-0.0.1
Status: Running
Pid: 275182
Port: 1019
LogConfigFile: /etc/vmware/vmdkops/log_config.json
LogFile: /var/log/vmware/vmdk_ops.log
LogLevel: INFO
Running unit tests in /tmp/vmdk_ops_unittest18380/vsan_info_test.py...
ss
----------------------------------------------------------------------
Ran 2 tests in 0.000s

OK (skipped=2)
Running unit tests in /tmp/vmdk_ops_unittest18380/vsan_policy_test.py...
sssss
----------------------------------------------------------------------
Ran 5 tests in 0.001s

OK (skipped=5)

```

### CODE REVIEW

File: esx_service/vmdk_ops_test.py [hunk 1]
```diff
@@ -380,7 +388,10 @@ class VmdkAttachDetachTestCase(unittest.TestCase):
 
     volNamePre = "vol_UnitTest_Attach"
     vm_name = 'test-vm'
-    max_vol_count = 60
+    if config["run_max_attach"]:
+        max_vol_count = MAX_VOL_COUNT_FOR_ATTACH
```
Comment (pdhamdhere) [inline esx_service/vmdk_ops_test.py]: Why MAX_VOL_COUNT_FOR_ATTACH? We should either run_max_attach test or not run at all?

Comment (msterin) [inline esx_service/vmdk_ops_test.py]: Good point, we should use uniittest features to turn on/off tests. How about I drop this part from here and submit a separate PR with test selection?

Comment (pdhamdhere) [inline esx_service/vmdk_ops_test.py]: Fine with me.

**Comment by kerneltime** [comment]
The testbed should be up shortly but we will need to plat with drone a bit more to see the ideal setup.

**Comment by kerneltime** [comment]
If esx_service/vol_test.py was added but not run, we should first see if there is some thing of value that we should include.. why is it being removed?
